Task 3/Task 3.1/Task 3.1/Task 3.1/Classes/PeopleList.cs

```diff
@@ -0,0 +1,66 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Task_3._1.Classes
+{
+    public class PeopleList<T> : ICollection<T>
+    {
+        public List<T> People { get; set; }
+
+        public PeopleList(List<T> people)
+        {
+            People = people;
+        }
+
+        public int Count => People.Count;
+
+        public bool IsReadOnly => throw new NotImplementedException();
+
+        public void Add(T item)
+        {
+            People.Add(item);
+        }
+
+        public void Clear()
+        {
+            People.Clear();
+        }
+
+        public bool Contains(T item)
+        {
+            return People.Contains(item);
+        }
+
+        public void CopyTo(T[] array, int arrayIndex)
+        {
+            People.CopyTo(array, arrayIndex);
+        }
+
+        public IEnumerator<T> GetEnumerator()
+        {
+            for (int i = 0; i <= Count; i++)
+            {
+                if (i == Count)
+                {
+                    i = 0;
+                    yield return People[i];
+                }
+                else
+                    yield return People[i];
```

Comment:
You can move yield line out of if - this line is executed in every code branch, so we can move in out of branch like so
```suggestion
                if (i == Count) i = 0;
                
                yield return People[i];
```

Reply:
Don't know about it, thank you for the information!